packages/babylon-wallet-connector/src/core/wallets/btc/ledger/provider.ts

```diff
@@ -71,15 +105,30 @@ export class LedgerProvider implements IBTCProvider {
     if (!extendedPubKey) {
       throw new Error("Could not retrieve the extended public key for policy");
     }
+
     const networkDerivationIndex = this.getNetworkDerivationIndex();
-    const policy = new DefaultWalletPolicy("tr(@0/**)", `[${fpr}/86'/${networkDerivationIndex}'/0']${extendedPubKey}`);
+    const purpose = this.derivationConfig.purpose;
+
+    // Select policy template based on purpose
+    // purpose 86: Taproot (tr)
+    // purpose 84: Native SegWit (wpkh)
+    let policyTemplate: string;
+    if (purpose === 86) {
+      policyTemplate = "tr(@0/**)";
+    } else {
```

Comment:
Could be a good idea to do some defensive check here so that if purpose is not 86 or 84 we throw an error